### ISSUE TITLE
Confusion in some Codes

### ISSUE THREAD

**Issue by ArabianAce** [hashlips-lab/nft-erc721-collection#105]
Hi Liarco,
I am again here to take your time but i don't have any other option because I have a presale in 15 hours. I would rather compensate something if you share your wallet address: 

1) 
<img width="887" alt="image" src="https://user-images.githubusercontent.com/95887192/157595373-ffb679fe-31ee-4fa8-9ce0-3b7080138c1f.png">
The code in this image, the top one was setup for hashlips-lab however I replaced that with my co-owner address (Sorry for that :P). Is that a right way? We can do like this? My co-owner will get paid like this, right?
The below one, where it is written payable(owner()).call; Do we need to replace owner() with Owner's (Mine) wallet address or it is automatically calling for owner's payout in the default way?

Asking this because after deploying on Mainnet, I minted from my address (Owner). It had to return back to me (75%) and my co-owner (25%). But nothing returned to both of us. Though i was still at dev-server. Contract address: 0xdcC4933E5c97cEe592752efB6C32735Ce1132540

2) As i have a whitelist sale tonight, While on local dev server (Dapp section), I will give a command of up to yarn whitelist-open --network truffle and then control C to close and yarn build to upload on my server? or will i have to setup a full dapp then build it?

I would like to say thank you in advance.

**Comment by BlueGreninja** [comment]
@ArabianAce Yes you can do like this there is nothing wrong here, if you call the withdraw button then the address you set for your co-owner will get 25% and the owner wallet will get the remaining 75%
You just have to call the withdraw function in the write contract tab

**Comment by liarco** [comment]
Hi @ArabianAce,
you could add your co-owner as another split payment entry without removing the HashLips Lab's address and you could even lower the percentage if you wanted to (I just say that because it really helps maintaining all these projects which takes a lot of time and effort). Anyway you can also find my wallet address in [my GitHub profile page](https://www.github.com/liarco), I really appreciate the support from people using my tools.

That said, the funds stay on the contract until you run the `withdraw()` function (as @BlueGreninja just said correctly), so you should run that function anytime you want to move the funds from the contract to your wallets.

Let me know if this is clear enough and I wish you the best of luck with your project.

**Comment by ArabianAce** [comment]
Thank You @liarco and @BlueGreninja . I am still confused how to call the withdraw function? I just need to add my wallet address in withdraw under write tab? How about withdrawal to my co-owner's wallet in this case?

I am not a developer. Can't understand these things if not told me as a noob student. 

@liarco The contract already deployed, the 25% line is incorrect? these 25% will get stuck in this way and i will only be able to get 75% withdrawal every time or what? What are require(hs) and require(os) functions? How to redo this?
Because i did this, i had to ask your wallet address separately for the compensation. You deserve this. 

Also, Point 2 is not replied.

**Comment by liarco** [comment]
@ArabianAce please let me know if #107 answers your question about how to run the withdraw function.

When you (the owner) run the `withdraw()` function, then all payments in it will be processed. So, for example, you run that function once and both addresses receive their % (25 and 75). You can also run it as many times as you want during the sale, you don't have to wait 'till the end if you need it.

If you deployed the code above, then everything is fine, you don't have to deploy again.
The `require()` statements are just checking that the payment transactions completed successfully.

> Because i did this, i had to ask your wallet address separately for the compensation. You deserve this.

I really appreciate it, as I said you can check [my GitHub profile](https://www.github.com/liarco), I'm not pasting my address here so I don't have to update it everywhere in the future if I change it.

### About point 2
This should answer your question: https://github.com/hashlips-lab/nft-erc721-collection/issues/85#issuecomment-1061258188

**Comment by ArabianAce** [comment]
Thanks Again.
Yes I read #107 Means I just need to click the "Write" button shown in the image below whenever i want to withdraw? That's all?
<img width="1383" alt="image" src="https://user-images.githubusercontent.com/95887192/157675341-11b925ad-ae14-40a1-ad8b-2d4047641791.png">

Point 2) Now 1 more question has arisen in my mind after reading the steps you mentioned in #85 i.e. I already added 800 whitelist addresses and dapp is still on dev-server. The current command is whitelist-open. Now if I add few more addresses in whitelist.json, will the already running dapp on dev-server automatically update alongwith? If yes, i simply then enter "control C" in minting-dapp node and then "yarn build". now will it be only have a whitelist-open function to be hosted on my server? right?

And then when i will have my presale to open, i will have to redo this by closing whitelist and opening presale and again yarn build to go to my hosting server again? Am i understanding this thing right?

**Comment by ArabianAce** [comment]
Point 1 is all clarified and tested. Thank you so much. 

Point 2 clarification is still there though. Also, why everyone on hashlips discord saying they are seeing white screen when uploading dapp files in their servers? I wish i don't get this issue

**Comment by BlueGreninja** [comment]
> 

@ArabianAce If you are following everything as hashlips is doing then you won't get any problem, Whenever you change Whitelist.json file you have to run the command " yarn whitelist-open --network truffle " or " mainnet " Whichever your case might be. It will update the root hash, and yes the dapp will be updated automatically on the local host but if you have to put this code in hosting platform, then you have to do yarn build every time you make a change in the Dapp

Hope this helps

**Comment by ArabianAce** [comment]
@BlueGreninja Hashlips actually never showed how to add on to the public server. Daniel just told how to do that. But i will follow the same. 

All clarified. Thank to Both of you. Much appreciated.

**Comment by BlueGreninja** [comment]
> 

Glad to be of help, Wish you the best for your project!

**Comment by ArabianAce** [comment]
Thank you for wishing me the Luck :) 5 hours 35 mins to go for the whitelist sale

**Comment by ArabianAce** [comment]
Alright understood everything now. Thanks @liarco

**Comment by ArabianAce** [comment]
@liarco How to change these images in minting-dapp src folder? I want to replace these 3 images with my images. 

<img width="1608" alt="image" src="https://user-images.githubusercontent.com/95887192/157698857-38b20e55-ffd6-46dd-8872-e9b031fb30ed.png">

Shall I just delete these here and replace with my images with the same names? Am i right? Will these automatically be updated in build/images folder?

Thanks in Advance

**Comment by liarco** [comment]
@ArabianAce yes, you simply have to replace those images with other ones, but please stay on topic. Next time, if you have questions about unrelated things you should open a dedicated issue so it will be easier for people to search for the same thing in the future. 😉 Thank you!
